### ISSUE TITLE
storybook does not handle react-native@0.61.0 new fast refresh mode

### ISSUE THREAD

**Issue by mgcrea** [storybookjs/storybook#7916]
**Describe the bug**

Following the release of react-native@0.61.0-rc.0, legacy "hot module reloading" and "live reloading" have been merged in a single new "fast refresh" mode to hot reload components (more details in this [blog post](https://medium.com/@jonnyburger/first-look-react-native-0-61-with-fast-reloading-ad387e502e6f)).

Unfortunately, this new mode does not work with storybook as I get ` WARN  Story with id button--default-view already exists in the store!` errors when fast refresh kicks in.

Since livereload isn't an option anymore, it makes developing components with storybook highly impractical as you don't have any refresh anymore.

**To Reproduce**
1. Create a new react-native project with `react-native init AwesomeProject --version react-native@next`
2. Install storybook, and save/update a story
3. See warning, no refresh

**Expected behavior**
Working fast refresh while editing a story

**System:**
```
Environment Info:

  System:
    OS: macOS 10.14.5
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Browsers:
    Chrome: 76.0.3809.132
    Firefox: 68.0.1
    Safari: 12.1.1
  npmPackages:
    @storybook/react-native: ^5.1.11 => 5.1.11 
```

Note that I'm using the following hack to have working hooks:

```
addDecorator(Story => <Story />);
```

**Comment by shilman** [comment]
@mgcrea Any interest in trying to tackle this?

**Comment by mgcrea** [comment]
@shilman Yep, could try to fix this, but I'm not sure to know where to look,

The warning does seem purely informative (see [story_store.ts](https://github.com/storybookjs/storybook/blob/next/lib/client-api/src/story_store.ts#L186)).

Any insights on where I should look/fiddle?

**Comment by shilman** [comment]
@Gongreg @benoitdion any ideas?

**Comment by Gongreg** [comment]
@shilman, I will keep this task open and close the old one, since this format is replacing hot reloading.

**Comment by Gongreg** [comment]
Hi.
Actually fast reload does work, but it doesn't work when you are using `addDecorator(Story => <Story />)`
.

Edit:

I've played around a bit more and I see that if you are editing the same story you have visible the fast refresh works well. There are other issues, but I think they are related to: https://github.com/storybookjs/storybook/issues/8306

So If you are changing only something deep enough inside so that the top level rerender doesn't happen then you won't get a warning.

@shilman, I am not sure how are you going to handle full rerender on the web, when stories are updated?

**Comment by jarodevs** [comment]
Hello, so i am not actually using any hooks neither the addDecorator function. I still have a : 
`TypeError: JSON.strinfigy cannot serialize cyclic structures` when using fresh reload on RN.
Any fix to it?

Further description of the error at : https://github.com/storybookjs/storybook/issues/8734

**Comment by Gongreg** [comment]
Hey. To sum it up, react native storybook simply doesn't properly handle fast refresh at the moment. Even if it works a little bit it is just pure luck.

Everyone is welcome to try to make it work :)

**Comment by AlexandreLage** [comment]
Sadly I can't use this package then :'(

**Comment by Gongreg** [comment]
@AlexandreLage, sorry to hear that.
It can definitely be fixed, but someone has to work on it.

**Comment by AlexandreLage** [comment]
Hey, just `disableWebsocket: false` did the trick. 

```
const StorybookUIRoot = getStorybookUI({
  asyncStorage: AsyncStorage,
  port: 7007,
  host: "0.0.0.0",
  onDeviceUI: false,
  resetStorybook: true,
  disableWebsockets: true
});
```

**Comment by dbenfouzari** [comment]
By doing this `disableWebsockets: true` Storybook server does not work anymore

**Comment by stephanos7** [comment]
But disabling web sockets also cuts communication with the web server to preview stories on the browser correct? disableWebsockets: false allows me to render the stories in the iOS simulator but stories have stopped loading on the browser, is this expected with this workaround?

**Comment by dbenfouzari** [comment]
@stephanos7 try to insert option `onDeviceUI: false` to make it work again.

**Comment by Anitorious** [comment]
@shilman @dbenfouzari These work arounds only work if you don't intend to use the features of storybook that make it a good investment. As it stands, it's currently a glorified sandbox environment for static components. Tagging it with `has workaround` may be misleading? I'd like to help in anyway I can to get these issues resolved though.

**Comment by Gongreg** [comment]
@Anitorious, feel free to try to fix the issue. As it currently stands I see a lot of people who would benefit from having fast refresh working out of the box.

**Comment by stephanos7** [comment]
> @stephanos7 try to insert option `onDeviceUI: false` to make it work again.

@dbenfouzari thanks, this did load the available stories on the web's side bar - though nothing renders when you select a story. Not entirely sure if my issue is related to this, will dig into it.
